test/common/packagelib.py

```diff
@@ -87,6 +88,13 @@ def setUp(self):
             self.machine.execute(f"mkdir -p {empty_repo_dir} || true")
             self.machine.execute(f"repo-add {empty_repo_dir}/empty.db.tar.gz")
             self.machine.write("/etc/pacman.conf", config)
+            # Clean up possible leftover lockfile
+            self.machine.execute("""
```

Comment:
This is still a strange place to do it. This runs *after* restore_dir(), so it means that the lock file is in the real directory. This can clean it up, but it will be restored back after the test runs, so every subsequent test will try to remove it again.

I also still don't understand this in the first place -- unless the lock file is already present right after a fresh boot, the restore_dir() should already take care of removing lock files that got produced while a test runs. Why doesn't this happen?